File: backend/schemas/translation.py
"""Translation schemas for minutes and summary artifacts."""

from enum import StrEnum

from pydantic import BaseModel, Field


class TranslationSourceType(StrEnum):
    """Supported source artifact types for translation."""

    AUTO = "auto"
    MINUTES = "minutes"
    SUMMARY = "summary"


class TranslationCreateRequest(BaseModel):
    """Translation generation request."""

    target_language: str = Field(..., min_length=2, max_length=32)
    source_language: str | None = Field(default=None, min_length=2, max_length=32)
    source_type: TranslationSourceType = Field(default=TranslationSourceType.AUTO)
    max_tokens: int = Field(default=2400, ge=512, le=4096)
    force_refresh: bool = Field(default=False)


class TranslationResponse(BaseModel):
    """Generated or cached translation response."""

    task_id: str = Field(..., min_length=1)
    source_type: TranslationSourceType
    source_language: str | None = None
    target_language: str = Field(..., min_length=2)
    translated_text: str = Field(..., min_length=1)
    source_excerpt: str = Field(default="")
    cached: bool = Field(default=False)
    created_at: str = Field(..., description="ISO-8601 creation timestamp")

File: client/lib/models/translation.dart
class TranslationResult {
  final String taskId;
  final String sourceType;
  final String? sourceLanguage;
  final String targetLanguage;
  final String translatedText;
  final String sourceExcerpt;
  final bool cached;
  final String createdAt;

  const TranslationResult({
    required this.taskId,
    required this.sourceType,
    required this.sourceLanguage,
    required this.targetLanguage,
    required this.translatedText,
    required this.sourceExcerpt,
    required this.cached,
    required this.createdAt,
  });

  factory TranslationResult.fromJson(Map<String, dynamic> json) =>
      TranslationResult(
        taskId: json['task_id'] as String? ?? '',
        sourceType: json['source_type'] as String? ?? 'auto',
        sourceLanguage: json['source_language'] as String?,
        targetLanguage: json['target_language'] as String? ?? '',
        translatedText: json['translated_text'] as String? ?? '',
        sourceExcerpt: json['source_excerpt'] as String? ?? '',
        cached: json['cached'] as bool? ?? false,
        createdAt: json['created_at'] as String? ?? '',
      );
}
